hyperapp.d.ts

```diff
@@ -54,42 +59,48 @@ export type ActionResult<State extends Hyperapp.State> =
   | null
   | void
 
-export type ActionCallers<
+export interface Action<State extends Hyperapp.State> {
+  (data: any): ActionResult<State>
+}
+
+export interface Actions<State extends Hyperapp.State> {
+  // should be Actions<State[action]>, but this is not allowed by typescript
```

Comment:
Perhaps we could link to an issue with the TypeScript compiler so we can fix this in the future?

Reply:
@Mytrill

Reply:
Changed to:
``` typescript
export interface Actions<
  State extends Hyperapp.State & Partial<Record<keyof Actions<State>, any>>
> {
  [action: string]:
    | Actions<State[keyof Action<State, any>]>
    | Action<State, any>
}
```